Find a functionally equivalent code:android.text.DynamicLayout.updateBlocks:COMMENT
Method Modifier: hidden      
Comment:/**
 * This method is called every time the layout is reflowed after an edition.
 * It updates the internal block data structure. The text is split in blocks
 * of contiguous lines, with at least one block for the entire text.
 * When a range of lines is edited, new blocks (from 0 to 3 depending on the
 * overlap structure) will replace the set of overlapping blocks.
 * Blocks are listed in order and are represented by their ending line number.
 * An index is associated to each block (which will be used by display lists),
 * this class simply invalidates the index of blocks overlapping a modification.
 *
 * This method is package private and not private so that it can be tested.
 *
 * @param startLine the first line of the range of modified lines
 * @param endLine the last line of the range, possibly equal to startLine, lower
 * than getLineCount()
 * @param newLineCount the number of lines that will replace the range, possibly 0
 *
 * @hide
 */

@@ -38,46 +38,45 @@
         return;
     }
     if (newNumberOfBlocks > mBlockEndLines.length) {
-        final int newSize = ArrayUtils.idealIntArraySize(newNumberOfBlocks);
-        int[] blockEndLines = new int[newSize];
-        int[] blockIndices = new int[newSize];
+        int[] blockEndLines = ArrayUtils.newUnpaddedIntArray(Math.max(mBlockEndLines.length * 2, newNumberOfBlocks));
+        int[] blockIndices = new int[blockEndLines.length];
         System.arraycopy(mBlockEndLines, 0, blockEndLines, 0, firstBlock);
         System.arraycopy(mBlockIndices, 0, blockIndices, 0, firstBlock);
         System.arraycopy(mBlockEndLines, lastBlock + 1, blockEndLines, firstBlock + numAddedBlocks, mNumberOfBlocks - lastBlock - 1);

